src/app/scripts/cac/routing/cac-routing-itinerary.js

```diff
@@ -195,6 +197,48 @@ CAC.Routing.Itinerary = (function ($, cartodb, L, _, moment, Geocoder, Utils) {
             leg.formattedDuration = getFormattedDuration(leg);
             return leg;
         });
+        if (hasWaypoints) {
+            newLegs = mergeLegsAcrossWaypoints(newLegs);
+        }
+        return newLegs;
+    }
+
+    /* Waypoints always result in a step break, which ends up producing useless "To Destination"
+     * steps in the itinerary details.
```

Comment:
Regarding the comments on OTP's behavior: their treatment of waypoints as intermediate stops is intentional; the feature is there with the intent of the user adding extra stops to a trip (i.e. stop for gas), so the directions returned with extra explicit destinations make sense for that use case. What we're doing is routing via waypoints in order to create the appearance of live-editing a route, which is not an intended use for intermediate stops, though it serves our purpose well enough.